sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetSchemaPruning.scala

```diff
@@ -280,7 +280,7 @@ private[sql] object ParquetSchemaPruning extends Rule[LogicalPlan] {
           val leftFieldType = leftStruct(fieldName).dataType
           val rightFieldType = rightStruct(fieldName).dataType
           val sortedLeftFieldType = sortLeftFieldsByRight(leftFieldType, rightFieldType)
-          StructField(fieldName, sortedLeftFieldType)
+          StructField(fieldName, sortedLeftFieldType, nullable = leftStruct(fieldName).nullable)
```

Comment:
Agree with being conservative in general. For this fix, specifically, keeps the original nullability from the input as are, which is supposed to be. So, this is actually more conservative since it doesn't modify the input's nullability.

Reply:
Yes. It is also not good to have assumptions when we write a method like that. It makes the code coupled with outside context that the assumptions are based on. Ideally we should write methods which return correct results given just inputs.